pagination/page_request.go

```diff
@@ -0,0 +1,37 @@
+package pagination
+
+const defaultPageSize = 20
+
+type PageRequest struct {
+	page     int // page requested (default 1)
+	pageSize int // Number of items per page
+}
+
+func NewPageRequest(page, pageSize int) PageRequest {
+	if page < 1 {
+		page = 1
+	}
```

Comment:
I get your point. I think the behaviour is logical enough that it doesn't warrant an error. The library is also easier to use if there is no error generated.

However I will change this to return an error as you suggest (and update the tests). I'll stick to the standard library error because the errors are simple, we won't need to supply a context and we can avoid introducing a dependency.